test/src/kbasesearchengine/test/events/handler/WorkspaceEventHandlerTest.java

```diff
@@ -696,4 +703,528 @@ private void failLoadWSGetWSInfoException(final Exception toThrow, final Excepti
             TestCommon.assertExceptionCorrect(got, expected);
         }
     }
+
+    @Test
+    public void updateEventAccessGroupEvent() throws Exception {
+        final CloneableWorkspaceClient clonecli = mock(CloneableWorkspaceClient.class);
+        final WorkspaceClient cloned = mock(WorkspaceClient.class);
+        final WorkspaceClient wscli = mock(WorkspaceClient.class);
+        when(clonecli.getClientClone()).thenReturn(cloned);
+        when(clonecli.getClient()).thenReturn(wscli);
+
+        StatusEvent event = StatusEvent.getBuilder(
+                "WS",
+                           Instant.ofEpochMilli(20000),
+                           StatusEventType.PUBLISH_ACCESS_GROUP)
+                           .withNullableAccessGroupID(1)
+                           .build();
+
+        when(wscli.administer(argThat(new AdminGetWSInfoAnswerMatcher(1))))
+                .thenReturn(new UObject(wsTuple(1, "wsname", "username", "date", 7, "r", "n",
+                        "unlocked", Collections.emptyMap())));
+
+        final WorkspaceEventHandler weh = new WorkspaceEventHandler(clonecli);
+
+        // update event
+        StatusEvent updatedEvent = weh.updateObjectEvent(event);
+
+        // access group events are not updated, so we expect to get back the original event
+        Assert.assertEquals("expected the same event object", event, updatedEvent);
+    }
+
+    @Test
+    public void updateEventWorkspacePermanentlyDeleted() throws Exception {
+        final CloneableWorkspaceClient clonecli = mock(CloneableWorkspaceClient.class);
+        final WorkspaceClient cloned = mock(WorkspaceClient.class);
+        final WorkspaceClient wscli = mock(WorkspaceClient.class);
+        when(clonecli.getClientClone()).thenReturn(cloned);
+        when(clonecli.getClient()).thenReturn(wscli);
+
+        StatusEvent event = StatusEvent.getBuilder(
+                StorageObjectType.fromNullableVersion("WS", "foo", 1),
+                Instant.ofEpochMilli(20000),
+                StatusEventType.RENAME_ALL_VERSIONS)
+                .withNullableNewName("newName")
+                .withNullableAccessGroupID(1)
+                .withNullableObjectID("1")
+                .withNullableVersion(1)
+                .build();
+
+        when(wscli.administer(argThat(new AdminGetWSInfoAnswerMatcher(1))))
+                .thenThrow(new JsonClientException("No workspace with id 1 exists!"));
```

Comment:
is this the right message? 
```
In [5]: ws.get_workspace_info({'id': 31467})
---------------------------------------------------------------------------
*snip*

ServerError: JSONRPCError: -32500. Workspace 31467 is deleted
```

This is why I think a couple integration tests would be useful, so that if we mock something incorrectly it gets caught

Reply:
Looks like both messages are necessary I think

Reply:
Now that getWSInfo method is not being called anymore and only listObjects is being called, "Workspace 31467 is deleted" is the right message that is being checked and the only possible message for the exception case.

Reply:
Isn't this possible?
```
In [4]: ws.list_objects({'ids': [40000]})
---------------------------------------------------------------------------
ServerError                               Traceback (most recent call last)
*snip*
ServerError: JSONRPCError: -32500. No workspace with id 40000 exists
us.kbase.workspace.database.exceptions.NoSuchWorkspaceException: No workspace with id 40000 exists
```

Seems unlikely (at least while there's no workspace GC), but not totally impossible.